src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -9,6 +9,44 @@ public class RegistrationServiceImpl implements RegistrationService {
 
     @Override
     public User register(User user) {
+        if (userValidation(user)) {
+            return storageDao.add(user);
+        }
         return null;
```

Comment:
The `register` method should throw a custom unchecked exception when validation fails, instead of returning `null`. This change will provide clearer feedback on why registration failed, as per the task requirements.